File: sum_array.c
#include <stdio.h>

int sum_array(const int *arr, int length) {
    int sum = 0;
    for (int i = 0; i < length; i++) {
        sum += arr[i];
    }
    return sum;
}

int main(void) {
    int numbers[] = {1, 2, 3, 4, 5};
    int length = sizeof(numbers) / sizeof(numbers[0]);

    int result = sum_array(numbers, length);
    printf("Sum of array = %d\n", result);

    return 0;
}

File: sum_array.kt
fun sumArray(arr: IntArray): Int {
    var sum = 0
    for (num in arr) {
        sum += num
    }
    return sum
}

fun main() {
    val numbers = intArrayOf(1, 2, 3, 4, 5)
    val result = sumArray(numbers)
    println("Sum of array = $result")
}